src/gui/attributetable/qgsfeaturelistmodel.cpp

```diff
@@ -61,9 +61,13 @@ QVariant QgsFeatureListModel::data( const QModelIndex &index, int role ) const
     {
       return QSettings().value( "qgis/nullValue", "NULL" ).toString();
     }
+    else if ( role == QgsAttributeTableModel::FeatureIdRole )
+    {
+      return QVariant( 0 );
+    }
```

Comment:
This allows initializing the combobox to the _NULL_ item if the attribute value is NULL. Otherwise it would be initialized with the empty entry.